src/clusterfuzz/_internal/bot/fuzzers/centipede/engine.py

```diff
@@ -284,11 +301,44 @@ def fuzz(self, target_path, options, reproducers_dir, max_time):  # pylint: disa
     runner = _get_runner(target_path)
     _set_sanitizer_options(target_path)
     timeout = max_time + _CLEAN_EXIT_SECS
+
+    old_corpus_len = shell.get_directory_file_count(options.corpus_dir)
+    logs.info(f'Corpus length before fuzzing: {old_corpus_len}')
+
     fuzz_result = runner.run_and_wait(
         additional_args=options.arguments, timeout=timeout)
     log_lines = fuzz_result.output.splitlines()
     fuzz_result.output = Engine.trim_logs(fuzz_result.output)
 
+    workdir = options.workdir
+
+    corpus_minimization_failed = False
+    try:
+      time_for_minimize = timeout - fuzz_result.time_executed
+
+      self.minimize_corpus(
+          target_path=target_path,
+          arguments=[],
+          # New units, in addition to the main corpus units,
+          # are placed in new_corpus_dir. Minimize and merge back
+          # to the main corpus_dir.
+          input_dirs=[options.new_corpus_dir],
+          output_dir=options.corpus_dir,
+          reproducers_dir=reproducers_dir,
+          max_time=time_for_minimize,
+          # Use the same workdir that was used for fuzzing.
+          # This allows us to skip rerunning the fuzzing inputs.
+          workdir=workdir)
+    except Exception as e:
+      corpus_minimization_failed = True
+      logs.error(f'corpus minimization failed: {e}')
+
+    if corpus_minimization_failed:
+      # If we fail to minimize, fall back to moving the new units
+      # from the new corpus_dir to the main corpus_dir.
+      engine_common.move_mergeable_units(options.new_corpus_dir,
+                                         options.corpus_dir)
+
```

Comment:
Can you leave a TODO (and maybe a self-reminder) to change this to a warning if this becomes a problem caused by user code rather than by ClusterFuzz or centipede?

Reply:
Also, I think there's no need for the extra variable, just do the move in the except block.